modules/login/jsx/loginIndex.js

```diff
@@ -0,0 +1,311 @@
+          username: state.form.value.username,
+          password: state.form.value.password,
+        }),
+      })
+      .then((response) => {
+        if (response.ok) {
+          response.json().then((data) => {
+            // success - refresh page and user is logged in.
+            window.location.href = window.location.origin;
+          });
+        } else {
+          response.json().then((data) => {
+            if (data.error) {
+              const state = JSON.parse(JSON.stringify(this.state));
+              if (data.error === 'password expired') {
+                // password expired
+                state.component.expiredPassword = {
+                  message: 'Password expired for user.',
+                  username: state.form.value.username,
+                };
+                state.mode = 'expired';
+              } else {
+                // incorrect password
+                state.form.error.toggle = true;
+                state.form.error.message = data.error;
+              }
+              this.setState(state);
+            }
+          });
+        }
+      })
+      .catch((error) => {
+        console.error('Error! ' + error);
+      });
+  }
+
+  /**
+   * Set mode.
+   *
+   * @param {string} mode - set as mode.
+   */
+  setMode(mode) {
+    const state = JSON.parse(JSON.stringify(this.state));
+    state.mode = mode;
+    this.setState(state);
+  }
+
+  /**
+   * @return {DOMRect}
+   */
+  render() {
+    // Waiting for async data to load.
+    if (!this.state.isLoaded) {
+      return <Loader/>;
+    }
+    if (this.state.mode === 'login') {
+      const study = (
+        <div dangerouslySetInnerHTML={{__html: this.state.study.description}}/>
+      );
+      const error = this.state.form.error.toggle ? (
+        <StaticElement
+          text={this.state.form.error.message}
+          class={'col-xs-12 col-sm-12 col-md-12 text-danger'}
+        />
+      ) : null;
+      const login = (
+        <div>
+          <section className={'study-logo'}>
+            <img src={this.state.study.logo}
+                 alt={this.state.study.title}/>
+          </section>
+          <FormElement
+            name={'loginIndex'}
+            action={''}
+            fileUpload={'false'}
+            onSubmit={this.handleSubmit}
+          >
+            <TextboxElement
+              name={'username'}
+              value={this.state.form.value.username}
+              onUserInput={this.setForm}
+              placeholder={'Username'}
+              class={'col-sm-12'}
+              autoComplete={'username'}
+              required={true}
+            />
+            <PasswordElement
+              name={'password'}
+              value={this.state.form.value.password}
+              onUserInput={this.setForm}
+              placeholder={'Password'}
+              class={'col-sm-12'}
+              required={true}
+              autoComplete={'current-password'}
+              isPassword={true}
```

Comment:
I don't see where this prop gets used.

Reply:
@HenriRabalais thanks. I forgot to remove that prop when I changed the TextElement component to the PasswordElement component.

Reply:
@laemtl do you know if eslint has a flag for giving props that don't exist?

Reply:
Here is the list of available rules for react: https://github.com/yannickcr/eslint-plugin-react#list-of-supported-rules

Reply:
`forbidExtraProps`